File: Java/TP3-1/src/Estante.java
import java.util.ArrayList;

public class Estante {

	private String nome;
	private Categoria categoria;
	private ArrayList<Livro> livros;

	public Estante(String nome, Categoria categoria) {
		this.nome = nome;
		this.categoria = categoria;
		this.livros = new ArrayList<Livro>();
	}

	public String getNome() {
		return nome;
	}

	public Categoria getCategoria() {
		return categoria;
	}

	public ArrayList<Livro> getLivros() {
		return livros;
	}

	public void inserirLivro(Livro l) {
		livros.add(l);
	}

	public void removerLivro(Livro l) {
		livros.remove(l);
	}
	
	public boolean removerLivroNome (String nomeLivro) {
		for(Livro l : livros) {
			if(l.getNome().equalsIgnoreCase(nomeLivro)) {
				livros.remove(l);
				return true;
			}
		}
		return false;
	}
	
	public void mostrarEstante() {
		System.out.println("Estante: " + getNome());
		System.out.println("Categoria: " + getCategoria());
	}

	public void listarLivros() {
		for (Livro l : livros) {
			l.mostrarInfoLivro();
		}
	}

	public void listarAutores() {
		for (Livro l : livros)
			l.getAutor();
	}

	public int getQtde() {
		return livros.size();
	}

	public boolean isVazia() {
		return livros.size() == 0;
	}

	public boolean pesquisarLivro(String nomeLivro) {
		for (Livro l : livros) {
			if (l.getNome().equalsIgnoreCase(nomeLivro)) {
				return true;
			}
		}
		return false;
	}
	
}

File: Kotlin/TPK3-1/src/Estante.kt
import java.util.ArrayList

class Estante(nome: String, categoria: Categoria) {
	private var nome: String
	private var categoria: Categoria
	private var livros: ArrayList<Livro>
	var qtde: Int = 0

	fun getLivros(): ArrayList<String> {
		return livros.size()
	}

	fun getNome(): String {
		return nome
	}

	fun getCategoria(): Categoria {
		return categoria
	}

	var isVazia: Boolean = false
	
	fun getLivro(): ArrayList<String> {
		return livros.size()
	}

	init {
		this.nome = nome
		this.categoria = categoria
		this.livros = ArrayList<Livro>()
		this.qtde = qtde
	}

	fun inserirLivro(l: Livro) {
		livros.add(l)
	}

	fun removerLivro(l: Livro) {
		livros.remove(l)
	}

	fun removerLivroNome(nomeLivro: String): Boolean {
		for (l in livros) {
			if (l.getNome().equalsIgnoreCase(nomeLivro)) {
				livros.remove(l)
				return true
			}
		}
		return false
	}

	fun mostrarEstante() {
		println("Estante: " + nome)
		println("Categoria: " + categoria)
	}

	fun listarLivros() {
		for (l in livros) {
			l.mostrarInfoLivro()
		}
	}

	fun listarAutores() {
		for (l in livros)
			l.getAutor()
	}

	fun pesquisarLivro(nomeLivro: String): Boolean {
		for (l in livros) {
			if (l.getNome().equalsIgnoreCase(nomeLivro)) {
				return true
			}
		}
		return false
	}
}